homeassistant/components/fronius/sensor.py

```diff
@@ -206,6 +207,14 @@ class FroniusSensorEntityDescription(SensorEntityDescription):
         key="error_code",
         entity_category=EntityCategory.DIAGNOSTIC,
     ),
+    FroniusSensorEntityDescription(
+        key="error_message",
+        response_key="error_code",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.ENUM,
+        options=list(dict.fromkeys(INVERTER_ERROR_CODES.values())),
```

Comment:
```suggestion
        options=list(set(INVERTER_ERROR_CODES.values())),
```
I'm assuming the `dict.fromkeys` is to maintain uniqueness, you can also use a set for that

Reply:
`set()` doesn't preserve order. Snapshot tests would fail randomly then.

Reply:
Is it purely testing that is the concern? As in, there are helpers for that.

Reply:
Well, I also think it's better UX if the list items are consistent between restarts. Imho that's worth the 2 extra μs.

<img width="515" alt="Bildschirmfoto 2024-12-23 um 08 54 37" src="https://github.com/user-attachments/assets/fe2bc01c-6e77-4652-bc80-05c86d2482bd" />

Now it's ordered by error codes - which have some kind of namespaces. Not that most users will ever notice, but still a nice touch 😃
- 000 : no error
- 1xx : momentarily caused by grid
- 3xx : feed in
- 4xx : service required
- 5xx : acknowledge required 
...

Reply:
Just FMI, what are those helpers?

Reply:
`pytest_unordered` is what we use in our tests.